packages/abtest/src/index.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { useState, useEffect, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { CONTROL } from './constants';
+import {
+	getAndSetGroup,
+	getCachedGroup,
+	isActive,
+	recordABTestEvent,
+} from './utils';
+
+/**
+ * ABTest component.
+ *
+ * Use to randomly display one of two components to control and experiment groups.
+ *
+ * @param {Object} props Component props.
+ * @param {string} props.name Name of the experiment.
+ * @param {Object} props.control React component to display to control group.
+ * @param {Object} props.experiment React component to display to experiment group.
+ * @param {Function|null} props.onComplete Optional. Callback triggered once A/B Test has completed setup.
+ * @param {number} props.start Optional. A/B Test start timestamp.
+ * @param {number} props.end Optional. A/B Test end timestamp.
+ */
+const ABTest = ( {
+	name,
+	control,
+	experiment,
+	onComplete,
+	start = 0,
+	end = Infinity,
+} ) => {
+	const [ variation, setVariation ] = useState( CONTROL );
+	const [ isFetching, setIsFetching ] = useState( true );
+	const active = isActive( start, end );
+	const handleComplete = useCallback( () => {
+		setIsFetching( false );
+		if ( onComplete ) {
+			onComplete();
+		}
+	}, [ onComplete ] );
+
+	useEffect( () => {
+		if ( ! active ) {
+			handleComplete();
+			return;
+		}
+
+		const cachedGroup = getCachedGroup( name );
+		if ( cachedGroup ) {
+			setVariation( cachedGroup );
+			handleComplete();
+			recordABTestEvent( name, cachedGroup, 'from_cache', 'serve' );
+		} else {
+			( async () => {
+				const group = await getAndSetGroup( name );
+				setVariation( group );
```

Comment:
Should this be `setGroup` to keep it all straight? I see this means the value in state must also be `group`, so maybe the const declared two lines above could be `groupFromDB` or something.

Reply:
Good call. I made this `variation`/`setVariation` to distinguish between component state and util functions. But looking at this again, it makes more sense to make this consistent. I've renamed these and changed the const you mentioned to `newGroup`: https://github.com/woocommerce/woocommerce-admin/pull/6536/commits/fd9021dcf1c2f4a9e1c16078604abf56171cac64